src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -76,7 +80,8 @@ func (r *StatefulSetResource) Ensure(ctx context.Context) error {
 	r.LastObservedState = &sts
 
 	// Ensure StatefulSet #replicas equals cluster requirement.
-	if sts.Spec.Replicas != r.pandaCluster.Spec.Replicas {
+	if sts.Spec.Replicas != nil && r.pandaCluster.Spec.Replicas != nil && *sts.Spec.Replicas != *r.pandaCluster.Spec.Replicas {
```

Comment:
we were comparing pointers so those differs all the time and we were patching the SS (or at least trying to patch it) all the time.